packages/style-engine/class-wp-style-engine.php

```diff
@@ -447,26 +487,75 @@ public function get_block_supports_styles( $block_styles, $options ) {
 		return $styles_output;
 	}
 
+	/**
+	 * Returns a stylesheet of CSS rules from a theme.json/global styles object.
+	 *
+	 * @param array $global_styles Styles object from theme.json.
+	 * @param array $options       array(
+	 *     'selector' => (string) When a selector is passed, the style engine will return a full CSS rule `$selector { ...rules }`, otherwise a concatenated string of properties and values.
+	 * );.
+	 *
+	 * @return string A stylesheet.
+	 */
+	public function generate_global_styles( $global_styles, $options ) {
+		if ( empty( $global_styles ) || ! is_array( $global_styles ) ) {
+			return null;
+		}
+
+		// The return stylesheet.
+		$global_stylesheet = '';
+
+		// Layer 0: Root.
+		$root_level_options_defaults = array(
+			'selector'          => 'body',
+			'custom_properties' => array(
+				'spacing' => array( 'blockGap' => '--wp--style--block-gap' ),
```

Comment:
An alternative to this is to allow merging of block definition metadata. I've tested that over here: https://github.com/WordPress/gutenberg/pull/40955/files#diff-f28aa79997745a4a51f917012b45a809c9b45d55134b55379a718f5c751dddf4R539

It works fine, but I'm hesitant to open up a way to modify the definitions without a bunch of checks, at least not before the style engine is more mature.